docs/source/conf.py

```diff
@@ -39,14 +39,16 @@
     'sphinx.ext.autosummary',
     'sphinx.ext.doctest',
     'sphinx.ext.intersphinx',
+    'sphinx.ext.ifconfig',
+    'sphinx.ext.extlinks',
```

Comment:
it's great to know these exist (I've been using extlinks feature for a while in scikit-learn but I never really knew what it was).

But I think  we should only introduce these if we have a use-case for it. I don't see a compelling use-case ATM for any of these, so I would suggest to revert, but maybe you had something in mind?

Reply:
One possible usecase I see for `ifconfig` is when we are classifying features into `alpha` `beta` and `prototype` we could avoid generating documentations for functions which are prototype use `ifconfig` 

`extlinks` might be useful for external links, but I see not much use-case though.
Happy to revert them though.

Reply:
Unless we have immediate use for them, I think we should avoid introducing them. We can introduce them when we need them :)